airflow/models/taskinstance.py

```diff
@@ -1135,8 +1136,63 @@ def _run_raw_task(
         if not test_mode:
             session.add(Log(self.state, self))
             session.merge(self)
+
         session.commit()
 
+        self._run_mini_scheduler_on_child_tasks(session)
```

Comment:
It may happened that we will do `rollback` on session we already committed, is is expected @jhtimmins ?